cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -345,13 +348,18 @@ public void generate(File inputFile, File inputProject, List<? extends Generable
                 report = cg.generate(input, finalTos, Paths.get(outputRootPath.getAbsolutePath()), false, classLoader,
                     templateFolder);
             }
-            if (ValidationUtils.checkGenerationReport(report) && isJavaInput) {
-                try {
-                    ParsingUtils.formatJavaSources(report.getGeneratedFiles());
-                } catch (FormatterException e) {
-                    LOG.info(
-                        "Generation was successful but we were not able to format your code. Maybe you will see strange formatting.",
-                        LOG.isDebugEnabled() ? e : null);
+            if (ValidationUtils.checkGenerationReport(report)) {
+                Set<Path> generatedJavaFiles = report.getGeneratedFiles().stream()
+                    .filter(e -> FileUtils.getExtension(e.toAbsolutePath().toString()).equals("java"))
+                    .collect(Collectors.toSet());
```

Comment:
As of performance, I would like to go with this approach: https://stackoverflow.com/a/25188854
You could even already reuse the Pattern in the above changes where already for the extension is checked for java / yml / yaml.

Reply:
I could absolutely implement this. But does pattern/matcher really improve the performance? The string compare is computational expensive?

Reply:
I think the compare is not an issue, but constructing the absolutepath or actually the path string of a long path is more expensive.

You could even think of asking path.getLastSegment().endsWith(".java"), which would even be very performant. Maybe even more.

Reply:
If it is not about the comparing, then either with pattern or contains/equals I do need to construct the absolute path. Unfortunately we are using java.nio.Path which does not have the getLastSegment() method